cou/steps/plan.py

```diff
@@ -304,10 +304,34 @@ async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan
     :rtype: UpgradePlan
     """
     pre_plan_sanity_checks(args, analysis_result)
-    hypervisors = await filter_hypervisors_machines(args, analysis_result)
-    logger.info("Hypervisors selected: %s", hypervisors)
     target = determine_upgrade_target(analysis_result)
 
+    plan = generate_common_plan(analysis_result, args, target)
+
+    control_plane_plan = await generate_control_plane_plan(analysis_result, args, target)
+    data_plane_plan = await generate_data_plane_plan(analysis_result, args)
+
+    for group in control_plane_plan + data_plane_plan:
+        if group:
```

Comment:
We have function to generate empty plan an here we are separeting filtering that, so like this we have two if instead of one and I do not see any benefit of it.

Reply:
yeah I find it awkward as well. One possible way around it would be to have `add_step` be able to handle noop plans, so you never have to worry about what you're sending through

Reply:
It's ok to have empty plan/step since it is not printed and not executed.